### PR TITLE
website rewrite

### PR DESCRIPTION
**Pull request by jacobdalamb**
Rewrite of website using [Starlight](https://starlight.astro.build/)

[Prettier](https://prettier.io/) and [Biome](https://biomejs.dev/) for formatting and linting.

- [ ] Update page descriptions

[github.io deployment info](https://docs.astro.build/en/guides/deploy/github/)

https://github.com/lycheeverse/lycheeverse.github.io/issues/63

## Preview Images
homepage
docs

### CODE REVIEW

**Comment by mre** [comment]
This looks pretty amazing!

Just tried it locally and it works great and is superfast.
Kudos for pulling this off @jacobdalamb.

I needed to add `https://` to the `site` parameter to make it run locally. So `site: "https://lychee.github.io",` in `astro.config.mjs`.

Also, I did an `npm install --save-dev astro` before I was able to launch it.

There are a couple of code cleanups necessary to remove some boilerplate I guess.
The landing page also needs some touching up, but that's optional and could also be done later. What I'd like to do there is to put the logo above the text.
But apart from that, we should be good to merge. Thanks!

**Comment by jacobdalamb** [comment]
I think https://pls.cli.rs/ is pretty cool website to compare to and get inspiration for the new site.

File: src/content/docs/guides/docker.md [hunk 1]
```diff
@@ -7,10 +10,10 @@ instead** 🐳.
 <img src="https://img.shields.io/docker/pulls/lycheeverse/lychee?color=%23099cec&logo=Docker" alt="Docker Pulls Badge" />
 </a>
 
-We provide a handy official image over at [lycheeverse/lychee][docker-image],
```
Comment (mre) [inline src/content/docs/guides/docker.md:10]: Does that mean link labels are not supported by Starlight?

File: src/content/docs/index.mdx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+title: Starlight 🌟 Build documentation sites with Astro
```
Comment (mre) [inline src/content/docs/index.mdx:2]:
```suggestion
title: lychee link checker
```

File: src/content/docs/index.mdx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+description: Starlight helps you build beautiful, high-performance documentation websites with Astro.
```
Comment (mre) [inline src/content/docs/index.mdx:3]:
```suggestion
description: A fast, async link checker written in Rust. Finds broken hyperlinks and mail addresses, supports Markdown, HTML and more, simple and lightweight.
```

File: src/content/docs/index.mdx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+template: splash
+editUrl: false
```
Comment (mre) [inline src/content/docs/index.mdx:5]: Is there a way to enable this? Would be cool to show an edit button.

File: src/content/docs/reference/example.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (mre) [inline src/content/docs/reference/example.md:1]: Is this file necessary or can it be deleted?

**Review by mre — COMMENTED** [review]
Added some comments. I think we can merge it after that.

**Comment by jacobdalamb** [comment]
> Added some comments. I think we can merge it after that.

I'll get the other suggestions later tonight (my time)

File: src/content/docs/guides/docker.md (hunk 1 above)
Comment (jacobdalamb) [reply src/content/docs/guides/docker.md:10]: Oh they are, I can put it back how it was, if that is easier for you? I removed the link labels because I like to keep to DRY (don't repeat yourself) code, but that's such a small thing.

File: src/content/docs/reference/example.md (hunk 5 above)
Comment (jacobdalamb) [reply src/content/docs/reference/example.md:1]: Removed.

File: src/content/docs/index.mdx (hunk 4 above)
Comment (jacobdalamb) [reply src/content/docs/index.mdx:5]: I setting it to `true` works.

File: src/content/docs/community.md [hunk 6]
```diff
@@ -1,8 +1,11 @@
-# Community
+---
+title: Community
+description: A guide in my new Starlight docs site.
```
[the remark this answers is not in the document]
Comment (jacobdalamb) [reply src/content/docs/community.md:3]: Removed.

**Comment by mre** [comment]
Okay, let's give it a go!
I've made some changes to add the necessary deployment config. 

Thanks so much for pulling this off @jacobdalamb. Great work and fast! 🚀

**Comment by mre** [comment]
The site is live!
Had to make a few last minute tweaks, but apart from that, it worked like a charm and is superfast! 😊
